R/multilevelannotationstep3.R

```diff
@@ -84,46 +97,27 @@ compute_score_pathways <- function(chemscoremat, matrix, pathwaycheckmode, score
               # if pathwaycheckmode is NOT "pm" is will be an error !!!
               num_chems <- round(num_chems, 0)
               
-              module_indices_in_pathway <- which(
-                chemscoremat$module_num == cur_module &
-                  chemscoremat$chemical_ID %in% pathway_chemicals &
-                  chemscoremat$score >= scorethresh &
-                  chemscoremat$Adduct %in% as.character(adduct_weights[, 1])
-              )
-              cur_module_data <- chemscoremat[module_indices_in_pathway, ]
-              a <- length(unique(cur_module_data$chemical_ID))
-              rm(cur_module_data)
-              
-              module_indices <- which(
-                chemscoremat$module_num == cur_module &
-                  chemscoremat$score >= scorethresh &
-                  chemscoremat$Adduct %in% as.character(adduct_weights[, 1])
-              )
-              cur_module_data2 <- chemscoremat[module_indices, ]
-              b <- length(unique(cur_module_data2$chemical_ID)) - a
-              rm(cur_module_data2)
-              
-              other_module_indices_in_pathway <- which(
-                chemscoremat$module_num != cur_module &
-                  chemscoremat$chemical_ID %in% pathway_chemicals &
-                  chemscoremat$score >= scorethresh &
-                  chemscoremat$Adduct %in% as.character(adduct_weights[, 1])
-              )
-              other_module_data <- chemscoremat[other_module_indices_in_pathway, ]
-              c <- length(unique(other_module_data$chemical_ID))
-              rm(other_module_data)
+              # a and b
+              module_indices <- which(chemscoremat$module_num == cur_module)
+              cur_module_data <- chemscoremat[module_indices, ]
+              result <- count_chemicals_occurence(cur_module_data, pathway_chemicals, scorethresh, adduct_weights)
+                
+              cur_module_number <- result$num_pathway
+              cur_module_pathway_number <- result$num
```

Comment:
This naming is confusing - I guess it should be named the other way around?

Reply:
Nice catch! Fixed in daa2941.